app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx

```diff
@@ -252,6 +306,49 @@ const JenkinsSetup = (): JSX.Element => {
 		fetchData();
 	}, [uuid, getServer]);
 
+	const setCopiedButtonStateToTrue = (copyButtonName: CopyButtonNameEnum) => {
```

Comment:
I think there's a better way to do this code. Let's have a chat over zoom :)

Reply:
What you're trying to do here is setting the next primary button. I'd call a `handleCopy` function on the button click to handle all actions, as there are likely a few.  I'd also use an array to determine priority of the buttons and help you easily find the next one. The beauty of something like this is that it scales easily to add/remove buttons, and reorder priority if we want to :) 

This is how I would recommend you go about this:

```
// buttons are stored in array in priority order. this is currently what you have in your state
const buttons = [
	{ name: 'buttonOne', clicked: false },
	{ name: 'buttonTwo', clicked: false },
	{ name: 'buttonThree', clicked: false },
	{ name: 'buttonFour', clicked: false },
	{ name: 'buttonFive', clicked: false }
];

// it's nice to have a `handleXyz` function to keep your jsx tidy
const handleCopy = ({ buttonName }) => {
	// do all the things you need to do when you handle a copy here


	// decide which button is next.
	// at this point in your code, you should've set the button to be `clicked: true`
	// please note I'm returning null here if all buttons have been clicked to prevent it from being set as undefined.
	const nextPrimaryButton =  buttons.find(button => button.clicked === false) || null;
	setNextPrimaryButton(nextPrimaryButton);
}
```

More details in our slack thread!